### ISSUE TITLE
'Reads' should not show up in the final `Resources:` section of a preview.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [pulumi/pulumi#2301]
A 'read' happens so we can get teh cloud state of a resource and then decide what we want to do with it.  it's an interim step as opposed to the result of comparing it against our current step.  

This is also problematic as 'reads' end up causing us to state that we're going to make change when we pass the test flag --expect-no-changes.

**Comment by CyrusNajmabadi** [comment]
I looked into this, and it's actually tough to accomplish on the UI layer.  That's because all we get back is a SummaryEvent, which just gives a count of steps per op-type.  So all we can know is "we did this many reads" not "we did N reads that resulted in no change, and M reads that did".  Talking to @swgillespie we could potentially address this on hte engine side.  The engine knows when it issues a read-step and then when it gets the result back for that step-event.  It could then translate the above two cases in teh summar so that a read-that-results-in-no-prop-changes gets recorded as an op-same, and one that results in changes gets recorded as an op-update.
